### PR TITLE
docs: use new doc generator

### PR DESCRIPTION
**Pull request by Foxandxss**
This is my proposed solution. A file at `demo/src` called `docs.json` that contains all the doc generated so from the demo's Angular, I can generate the doc.

I gitignored the file and I force a re-creation of the file always, so we don't end with stale information.

WDYT?

### CODE REVIEW

File: demo/src/app/components/accordion/accordion.component.ts [hunk 1]
```diff
@@ -3,17 +3,20 @@ import { Component } from '@angular/core';
...
 
 @Component({
   selector: 'ngbd-accordion',
   template: require('./accordion.component.html'),
   directives: [AccordionBasicComponent, ComponentDocsComponent, ExampleBoxComponent]
 })
 export class AccordionComponent {
-  accordionDocs = accordionDocs;
-  panelDocs = panelDocs;
+  accordionDocs = docs['NgbAccordion'];
```
Comment (wesleycho) [inline demo/src/app/components/accordion/accordion.component.ts]: Does `docs.NgbAccordion` not work?

Comment (icfantv) [inline demo/src/app/components/accordion/accordion.component.ts]: it should - i don't think any of our keys are non-standard.  but i admittedly have no preference here.

File: gulpfile.js [hunk 2]
```diff
@@ -19,7 +19,8 @@ var PATHS = {
   specs: 'src/**/*.spec.ts',
   demo: 'demo/**/*.ts',
   demoDist: 'demo/dist/**/*',
-  typings: 'typings/browser.d.ts'
+  typings: 'typings/browser.d.ts',
+  demoDocsJson: 'demo/src/docs.json'
```
Comment (icfantv) [inline gulpfile.js]: this is ok for now - we should either .gitignore this or move to a directory that's ignored so git doesn't keep pestering us that it's untracked.

Comment (Foxandxss) [inline gulpfile.js]: Yes, it is ignored by now.

Comment (pkozlowski-opensource) [inline gulpfile.js]: Could we move it to another folder, something that is specifically not under 'src'? Like `demo/temp` or `demo/generated`? This way people won't be tempted to generate + edit + change `.gitignore`.

**Comment by pkozlowski-opensource** [comment]
So my biggest grip with this approach is that I don't know how we are going to handle formatting of API description (ex. I would to highlight sth or just indicate `varName`). Shell we just use HTML directly in JSdocs? Markdown is sooooo convenient for those little things...

**Comment by pkozlowski-opensource** [comment]
At the same time let's not die over it - it is better to have sth working now - we can always change later. I'm going to pull it tomorrow and see how it works in practice with "real" content. Will report back.

File: demo/src/app/components/accordion/accordion.component.ts [hunk 3]
```diff
@@ -3,17 +3,20 @@ import { Component } from '@angular/core';
...
 export class AccordionComponent {
-  accordionDocs = accordionDocs;
-  panelDocs = panelDocs;
+  accordionDocs = docs.NgbAccordion;
+  panelDocs = docs.NgbPanel;
   basicHtmlContent = basicHtmlContent;
   basicTsContent = basicTsContent;
+
+  constructor() {
+    console.log(this.accordionDocs);
```
Comment (pkozlowski-opensource) [inline demo/src/app/components/accordion/accordion.component.ts]: remove

File: demo/src/app/components/accordion/accordion.component.html [hunk 4]
```diff
@@ -1,8 +1,8 @@
-<ngbd-component-docs [name]="accordionDocs.name"
+<ngbd-component-docs [name]="accordionDocs.selector"
```
Comment (pkozlowski-opensource) [inline demo/src/app/components/accordion/accordion.component.html]: We should have 2 separate things:
* name (class name) => people need to list it in `[directives]`
* selector

Comment (Foxandxss) [inline demo/src/app/components/accordion/accordion.component.html]: We have no "name" right now on the output file.

Comment (pkozlowski-opensource) [inline demo/src/app/components/accordion/accordion.component.html]: We do have `className`. Oh, while we are at it -> we should probably also use `fileName` to:
* let people know how to import this thing
* have a link to sources on GH

Comment (Foxandxss) [inline demo/src/app/components/accordion/accordion.component.html]: I don't think we want to recommend people to import concrete components by `fileName`.

Better to output scoped packages.

**Pull request closed without merging**